src/model/table_metadata.py

```diff
@@ -0,0 +1,34 @@
+from model.index import *
+
+
+class TableMetaData:
+    def __init__(self, table_schema, table):
+        TableMetaData.__validate__(table_schema)
+        self.__table_schema__ = table_schema
+        self.table_name = table_schema[Keys.NAME]
+        self.primary_key = table_schema[Keys.PRIMARY_KEY]
+        self.columns = table_schema[Keys.COLUMNS]
+        self.__path__ = table.get_path()
+        self.indices = []
+        for index in table_schema[Keys.INDEX_KEYS]:
+            self.indices.append(Index(index, self))
+
+    def get_path(self):
+        return self.__path__
+
+    def serialize(self):
+        self.__create_table_schema__()
+        self.__serialize_indices__()
+
+    @staticmethod
+    def __validate__(table_schema):
+        if table_schema[Keys.PRIMARY_KEY] is None or table_schema[Keys.PRIMARY_KEY] not in table_schema[Keys.COLUMNS]:
+            raise WrongParameterError("Primary_key not found")
+
+    def __create_table_schema__(self):
```

Comment:
This can be static and pass an object to it

Reply:
it is better to serialize `self.__dict__`  because later you will need this. 
Think whyyy???

Reply:
can you explain more

Reply:
You need to serialize this object `self` which is an instance of `TableMetadata`.
 In your next task, you will need to load this object.
You can load it easily by 

```
j = '{"action": "print", "method": "onData", "data": "Madan Mohan"}'
 import json

class Payload(object):
    def __init__(self, j):
         self.__dict__ = json.loads(j)
     p = Payload(j) 
```

https://stackoverflow.com/questions/15476983/deserialize-a-json-string-to-an-object-in-python